### PR TITLE
Project: Private events

### PR DESCRIPTION
**Pull request by OlukaDenis**
Complete all the implementations
- Authentication
- Associations

### CODE REVIEW

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Very nice work on this project, however, the following changes need to be done before moving on ⬇️** 

#### Tests

* [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [ ] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [ ] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Comment by OlukaDenis** [comment]
Added the Tests

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello @OlukaDenis 

Thank for the improvements but it's not sufficient. Take a lot on these screenshots
Screenshot_2020-02-18_20-18-35
Screenshot_2020-02-18_20-18-59

Also, you didn't write any integration tests. Please, read [that article](https://thoughtbot.com/blog/rspec-integration-tests-with-capybara) to learn more about writing integrations tests.

Kindly make the changes and submit a new code review request.
Thanks

Mael FOSSO

**Comment by OlukaDenis** [comment]
Thanks for the article but the tests run smoothly form my end.

File: app copy/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<h1>Members Posts</h1>
+
+<div class= "row">
+ <div class="col-md-6 col-md-offset-3">
+    <% if logged_in? %>
+      <%= link_to "New Post", new_post_path, class: 'post-new btn btn-primary' %>
+    <% end %>
+
+    
+    <%# <img src="img_avatar3.png" alt="John Doe" class="mr-3 mt-3 rounded-circle" style="width:60px;"> %>
```
Comment (mcrd25) [inline app copy/views/posts/index.html.erb:10]: Please remove commented out code. 
As can be seen from [this screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)
It is better to delete commented code. If you need to review a previous version of your code, use git history. 🙂

File: spec/controllers/events_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  describe 'GET #new' do
+    it { expect(response).to have_http_status(:success) }
+  end
+
+  describe 'GET #index' do
+    it { expect(response).to have_http_status(:success) }
+  end
+
+  # describe 'POST #create' do
```
Comment (mcrd25) [inline spec/controllers/events_controller_spec.rb:22]: Please remove commented out code. 
As can be seen from [this screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)
It is better to delete commented code. If you need to review a previous version of your code, use git history. 🙂

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  describe 'GET #new' do
+    it { expect(response).to have_http_status(:success) }
+  end
+
+  describe 'GET #create' do
+    it { expect(response).to have_http_status(:success) }
+  end
+
+  # describe 'POST #create' do
```
Comment (mcrd25) [inline spec/controllers/users_controller_spec.rb:17]: Please remove commented out code. 
As can be seen from [this screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)
It is better to delete commented code. If you need to review a previous version of your code, use git history. 🙂

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 👀 

Hi Denis! 👋 

Good job with the tests! There are only two minor things left to get the 👍 

#### Required Changes
- Remove all commented out code as can be seen from review comments
- When I click on "Happening" (the nav header) I get following error:
Screen Shot 2020-02-19 at 10 02 09 AM

#### Additional Suggestions [OPTIONAL]
- [OPTIONAL] Consider adding a link to profile in nav: Can only access profile by going to route directly besides when I initially log in.


We are almost there! Had it not been for that one bug I'd have given the 👍 Almost to the 🏁 

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by wathigo — APPROVED** [review]
### Changes Approved :heavy_check_mark: 
Hi Denis :wave: 
Great work implementing the suggested changes

**You may proceed to the next project.**

**Good luck :four_leaf_clover:** 
**Simon Wathigo**

**Pull request merged**